### PR TITLE
[config] Add commands to remove BGP neighbor configuration

### PR DESCRIPTION
**Pull request by rvisnu**
**- What I did**
I have added commands to remove BGP neighbor configuration
**- How I did it**
Added some new methods to the config/main.py file to support removing BGP neighbor configuration.
**- How to verify it**
```
admin@Test_Device1:~$ show ip bgp summary
BGP router identifier 10.10.1.1, local AS number 65534
RIB entries 1, using 112 bytes of memory
Peers 42, using 191 KiB of memory
Peer groups 1, using 56 bytes of memory

Neighbor        V         AS MsgRcvd MsgSent   TblVer  InQ OutQ Up/Down  State/PfxRcd
... 
10.10.193.43    4 64521       0       0        0    0    0 never    Idle       
10.10.193.45    4 64522       0       0        0    0    0 never    Idle       
10.10.193.47    4 64523       0       0        0    0    0 never    Idle       
10.106.8.240    4 64524       0       0        0    0    0 never    Idle       
10.106.12.240   4 64525       0       0        0    0    0 never    Idle       

Total number of neighbors 21

admin@Test_Device1:~$ redis-cli -n 4 hgetall "BGP_NEIGHBOR|10.106.12.240"        
 1) "rrclient"
 2) "0"
 3) "name"
 4) "Test_T2"
 5) "local_addr"
 6) "10.106.12.241"
 7) "nhopself"
 8) "0"
 9) "holdtime"
10) "10"
11) "asn"
12) "64525"
13) "keepalive"
14) "3"
admin@Test_Device1:~$ 

admin@Test_Device1:~$ 
admin@Test_Device1:~$ sudo config bgp remove neighbor 10.106.12.240
Removed configuration of BGP neighbor 10.106.12.240
admin@Test_Device1:~$ 

admin@Test_Device1:~$ redis-cli -n 4 hgetall "BGP_NEIGHBOR|10.106.12.240" 
(empty list or set)
admin@Test_Device1:~$ 

admin@Test_Device1:~$ show ip bgp summary 
BGP router identifier 10.10.1.1, local AS number 65534
RIB entries 1, using 112 bytes of memory
Peers 42, using 191 KiB of memory
Peer groups 1, using 56 bytes of memory

Neighbor        V         AS MsgRcvd MsgSent   TblVer  InQ OutQ Up/Down  State/PfxRcd
... 
10.10.193.43    4 64521       0       0        0    0    0 never    Idle       
10.10.193.45    4 64522       0       0        0    0    0 never    Idle       
10.10.193.47    4 64523       0       0        0    0    0 never    Idle       
10.106.8.240    4 64524       0       0        0    0    0 never    Idle              

Total number of neighbors 20

-------------

admin@Test_Device1:/usr/lib/python2.7/dist-packages$ show ipv6 bgp summary 
BGP router identifier 10.10.1.1, local AS number 65534
RIB entries 1, using 112 bytes of memory
Peers 42, using 191 KiB of memory
Peer groups 1, using 56 bytes of memory

Neighbor        V         AS MsgRcvd MsgSent   TblVer  InQ OutQ Up/Down  State/PfxRcd
...
2603:10b0:b0f:346::4e
                4 65515       0       0        0    0    0 never    Active     
2603:10b0:b0f:346::52
                4 65516       0       0        0    0    0 never    Active     
2603:10b0:b0f:346::56
                4 65517       0       0        0    0    0 never    Active     
2603:10b0:b0f:346::5a
                4 65518       0       0        0    0    0 never    Active     
2603:10b0:b0f:346::5e
                4 65519       0       0        0    0    0 never    Active     

Total number of neighbors 21

admin@Test_Device1:~$ redis-cli -n 4 hgetall "BGP_NEIGHBOR|2603:10b0:b0f:346::5e"
 1) "rrclient"
 2) "0"
 3) "name"
 4) "TestT0"
 5) "local_addr"
 6) "2603:10b0:b0f:346::5d"
 7) "nhopself"
 8) "0"
 9) "holdtime"
10) "10"
11) "asn"
12) "65519"
13) "keepalive"
14) "3"
admin@Test_Device1:~$ 
admin@Test_Device1:~$ 
admin@Test_Device1:~$ sudo config bgp remove neighbor 2603:10b0:b0f:346::5e
Removed configuration of BGP neighbor 2603:10b0:b0f:346::5e
admin@Test_Device1:~$ 
admin@Test_Device1:~$ redis-cli -n 4 hgetall "BGP_NEIGHBOR|2603:10b0:b0f:346::5e"
(empty list or set)
admin@Test_Device1:~$ 

admin@Test_Device1:/usr/lib/python2.7/dist-packages$ show ipv6 bgp summary 
BGP router identifier 10.10.1.1, local AS number 65534
RIB entries 1, using 112 bytes of memory
Peers 42, using 191 KiB of memory
Peer groups 1, using 56 bytes of memory

Neighbor        V         AS MsgRcvd MsgSent   TblVer  InQ OutQ Up/Down  State/PfxRcd
...
2603:10b0:b0f:346::4e
                4 65515       0       0        0    0    0 never    Active     
2603:10b0:b0f:346::52
                4 65516       0       0        0    0    0 never    Active     
2603:10b0:b0f:346::56
                4 65517       0       0        0    0    0 never    Active     
2603:10b0:b0f:346::5a
                4 65518       0       0        0    0    0 never    Active      

Total number of neighbors 20

-------------
admin@Test_Device1:~$ 
admin@Test_Device1:~$ show run bgp | grep Test_Device0
 neighbor 10.10.193.31 description Test_Device0
 neighbor 2603:10b0:b0f:346::42 description Test_Device0
admin@Test_Device1:~$ 

admin@Test_Device1:~$ 
admin@Test_Device1:~$ sudo config bgp remove neighbor Test_Device0
Removed configuration of BGP neighbor 2603:10b0:b0f:346::42
Removed configuration of BGP neighbor 10.10.193.31
admin@Test_Device1:~$ 
admin@Test_Device1:~$ 
```
**- Previous command output (if the output of a command-line utility has changed)**
The output on the command doesn't change. It just removes the neighbor details when you run the "show ip bgp summary" / "show ipv6 bgp summary" command
**- New command output (if the output of a command-line utility has changed)**
The output on the command doesn't change. It just removes the neighbor details when you run the "show ip bgp summary" / "show ipv6 bgp summary" command
-->

### CODE REVIEW

File: config/main.py [hunk 1]
```diff
@@ -952,6 +964,29 @@ def neighbor(ipaddr_or_hostname, verbose):
...
+#
+
+@bgp.group()
+def remove():
+    "Remove BGP neighbor configuration from the device"
+    pass
+
+@remove.command('neighbor_ip')
+@click.argument('neighbor_ip', metavar='<neighbor_ip>', required=True)
+def remove_neighbor_with_ip(neighbor_ip):
```
Comment (jleveque) [inline config/main.py]: Do you prefer to have separate commands for neighbor IP and neighbor name?

For shutting down/starting up BGP sessions (`config bgp shutdown neighbor ...` / `config bgp startup neighbor ...`), I allow the user to enter either the neighbor's IP address *or* hostname without the `neighbor_ip` or `neighbor_name` qualifiers. You can see how I did it here: https://github.com/Azure/sonic-utilities/blob/a134fd13a80415ce41f7afbe85d571aa9824f615/config/main.py#L941

Would you rather match the same approach here?

Comment (rvisnu) [reply config/main.py]: Thanks @jleveque.
submitted another iteration with single command to accommodate either ip address or hostname

File: config/main.py [hunk 2]
```diff
@@ -238,6 +238,31 @@ def _change_bgp_session_status(ipaddr_or_hostname, status, verbose):
     for ip_addr in ip_addrs:
         _change_bgp_session_status_by_addr(ip_addr, status, verbose)
 
+def _validate_bgp_neighbor_ip(neighbor):
```
Comment (jleveque) [inline config/main.py]: Now that the function can take either an IP address or hostname, I believe this function name and parameter name should be changed to be more clear. Maybe something like:

`def _validate_bgp_neighbor(ipaddr_or_hostname):`

Comment (rvisnu) [reply config/main.py]: changed as suggested in next iteration

Comment (jleveque) [reply config/main.py]: You changed the parameter name, but not the function name. Now that the function can take either an IP address or hostname, you should remove "ip" from the function name, as in my example above, `_validate_bgp_neighbor()`

Comment (rvisnu) [reply config/main.py]: Hi @jleveque, 
Renamed the method name. 
I can see the test failed but it is not related to my change.
Kindly help in retest the build

**Comment by jleveque** [comment]
Retest this please

**Comment by jleveque** [comment]
Retest this please

**Comment by jleveque** [comment]
Retest this please

**Comment by jleveque** [comment]
Retest this please

**Comment by rvisnu** [comment]
Retest this please

**Pull request merged**